src/cli/rustup_mode.rs

```diff
@@ -79,7 +92,7 @@ pub fn main() -> Result<()> {
             (_, _) => unreachable!(),
         },
         ("run", Some(m)) => run(cfg, m)?,
-        ("which", Some(m)) => which(cfg, m)?,
+        ("which", Some(m)) => which(cfg, m, toolchain)?,
```

Comment:
Why not just override `toolchain` in `cfg` and pass that in instead? We can keep the current version that way without adding a new argument.

Reply:
That is a very good suggestion.  If the config toolchain gets reset then all commands will benefit since it's effectively altering the default toolchain entry from the config file.  One thing which will have to be checked if we do that is that we don't end up saving the different toolchain back if we do something (not expected, but possible) along the lines of `rustup +another set profile default`  -> alters both `profile` and `default_toolchain` in `~/.rustup/settings.toml`

Reply:
@kinnison Ah, I did not though about that though. Maybe that should be added to the tests as well.

@BeniCheni Wow, fast.